src/Common/Core/Test/Fixtures/CoreShellProviderFixture.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.ComponentModel.Composition.Hosting;
+using System.Threading.Tasks;
+using Microsoft.Common.Core.Extensions;
+using Microsoft.Common.Core.Services;
+using Microsoft.Common.Core.Shell;
+using Microsoft.Common.Core.Test.Fakes.Shell;
+using Microsoft.UnitTests.Core.Shell;
+using Microsoft.UnitTests.Core.XUnit;
+using Microsoft.UnitTests.Core.XUnit.MethodFixtures;
+using Xunit.Sdk;
+
+namespace Microsoft.Common.Core.Test.Fixtures {
+    public abstract class CoreShellProviderFixture : MethodFixtureBase, ICoreShellProvider {
```

Comment:
`ICoreShell` isn't a service provider, `IServiceContainer` is. On some levels, `ICoreShell` is actually undesired. For example, `RSession` shouldn't have access not only to the `ICoreShell`, but to the `IUIService` as well. And so are the tests.

It it good to have `IServiceContainer` as an entry point all around, not only because it allows us to access any required service, but also because it allow scoping and doesn't bound existing code to a set of services that is defined by `IServiceContainer` implementation. And `ICoreShell` is actually just one of the services. It is an entry point for the application when it starts, but not for the rest of the code.